File: Java/src/main/java/org/codecop/lcdnumbers/NumeralSystemProvider.java
package org.codecop.lcdnumbers;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class NumeralSystemProvider {

    @Value("${numeral_systen_base}")
    private int base = 10;

    @Bean
    public NumeralSystem createNumeralSystem() {
        return new NumeralSystemImpl(base);
    }
}

File: Csharp-NAutowired/LcdNumbers/NumeralSystemProvider.cs
using Microsoft.Extensions.Options;
using NAutowired.Core.Attributes;

namespace Org.Codecop.Lcdnumbers
{
    public class NumeralSystemProvider
    {
        [Autowired]
        private IOptions<NumberBaseConfig> @base { get; set; }

        public INumeralSystem CreateNumeralSystem()
        {
            return new NumeralSystem(@base.Value.Base);
        }
    }

    public class NumberBaseConfig
    {
        public int Base { get; set; }
    }

}
